geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -607,6 +650,84 @@ private void addIfMatching(GlobPattern matchPattern, List<byte[]> resultList, by
     }
   }
 
+  private RedisSortedSet getIntersection(List<RedisSortedSet> sets, ZAggregator aggregator) {
+    RedisSortedSet retVal = new RedisSortedSet(Collections.emptyList(), new double[] {});
+    RedisSortedSet smallestSet = sets.get(0);
+
+    for (RedisSortedSet set : sets) {
+      if (set.getSortedSetSize() < smallestSet.getSortedSetSize()) {
+        smallestSet = set;
+      }
+    }
+
+    for (byte[] member : smallestSet.members.keySet()) {
+      Double newScore;
+      if (aggregator.equals(ZAggregator.SUM)) {
+        newScore = getSumOfScoresForMember(sets, member, retVal);
+      } else if (aggregator.equals(ZAggregator.MAX)) {
+        newScore = getMaxScoreForMember(sets, member, retVal);
+      } else {
+        newScore = getMinScoreForMember(sets, member, retVal);
+      }
+
+      if (newScore != null) {
+        if (newScore.isNaN()) {
+          throw new ArithmeticException(ERROR_OPERATION_PRODUCED_NAN);
+        }
+        retVal.memberAdd(member, newScore);
+      }
+    }
+    return retVal;
+  }
+
+  private Double getSumOfScoresForMember(List<RedisSortedSet> sets, byte[] member,
+      RedisSortedSet retVal) {
+    double runningTotal = 0;
+    for (RedisSortedSet set : sets) {
+      if (set.members.containsKey(member)) {
+        runningTotal += set.members.get(member).score;
```

Comment:
use getScore()
Also instead of calling both containsKey and get (which does two map lookups) just call get and if its result is not null compute the runningTotal.